File: Angular/src/app/BusinessObject/tag.ts
export type RawTag = {
  tagId: number,
  name: string,
  description: string,
}

export class Tag {
  tagId: number;
  name: string;
  description: string;

  constructor(rawData: Partial<RawTag>) {
    const {
      tagId = 0,
      name = '',
      description = '',
    } = rawData;

    this.tagId = tagId;
    this.name = name;
    this.description = description;
  }
}

File: DotNet/Api/Models/Tag.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Api.Models;

public class Tag
{
    [Key]
    public int TagId { get; set; }

    [MinLength(1)]
    public string Name { get; set; } = "";
    public string Description { get; set; } = "";

    [JsonIgnore]
    public int SongId { get; set; }

    [JsonIgnore]
    public Song Song { get; set; }
}